docs/ruleset_functions/get_proposed_hvac_modeled_with_virtual_heating.md

```diff
@@ -0,0 +1,50 @@
+            # Below compares the heating systems associated with the hvac system across the P_RMR and U_RMR only if heating was modeled in the proposed. If heating was modeled in the proposed and NOT in the U_RMR (the heating system type is None in the U_RMR) then this means virtual heating was modeled in the proposed. We are not checking the preheat coils because if no heating is specified in the design then the proposed should be modeled identically to the baseline and preheat coils are not applicable to any baseline systems.
+            - For each heating_system_p in hvac_p: `for heating_system_p in hvac_p:`
+                - Check if "None" is NOT the heating_system_type for heating_system_p, if its not "None" (means heating was modeled) then get the analogous heating system in the U_RMR and check if the heating_system_type is "None" in the U_RMR: `if heating_system_p.heating_system_type != "None":`
+                    - Get analogous heating system in U_RMR: `heating_system_u = match_data_element(U_RMR,HeatingSystem,heating_system_p.id)`
+                    - Check if the analogous system in the U_RMR has a heating system type equal to "None", if it does then set the virtual_heating_modeled_check_p boolean variable to TRUE : `if heating_system_u.heating_system_type == "None": virtual_heating_modeled_check_p = TRUE`
+            - Check if heating was modeled in the proposed via virtual_heating_modeled_check_p boolean variable (and not in the U_RMR) add hvac system to list: `if virtual_heating_modeled_check_p == TRUE: proposed_hvac_modeled_with_virtual_heating_list_p = proposed_hvac_modeled_with_virtual_heating_list_p.append(hvac_p.id)`
+    # Below checks if a terminal unit exists in both the P_RMR and U_RMR. If there is a mismatch then the HVAC system associated with the terminal unit is added to the list of HVAC systems in which virtual heating was modeled in the proposed. If the terminal unit exists in both then it is checked to see if both have heating. If there is a mismatch the HVAC system associated with the terminal unit is added to the list of HVAC systems with virtual heating modeled in the proposed.
+    - For each terminal unit in the P_RMR: `for each terminal_p in P_RMR..Terminal:` 
+        - Reset virtual_heating_modeled_check_p boolean variable: `virtual_heating_modeled_check_p = FALSE`
+        - Get analogous terminal unit in the U_RMR: `terminal_u = match_data_element(U_RMR,TERMINAL,terminal_p.id)`
+        - Check if null was returned: `if terminal_u == Null:` (waiting from group to hear about what is returned if there is no analogous terminal unit in the U_RMR, the alternative is to create a list of all terminal unit IDs in the U_RMR)
+            - Add the associated HVAC system to list with virtual heating:`proposed_hvac_modeled_with_virtual_heating_list_p.append(terminal_p.served_by_heating_ventilation_air_conditioning_system)`
+        - Else, null NOT returned: `Else:`
```

Comment:
@supriyagoel @mkarpman  Maria and I discussed this and we were thinking that for MZ baseline systems the heating in the baseline would be reheat which would be associated with the terminal unit and so since when a virtual system needs to be modeled in the proposed it needs to be modeled the same as in the baseline and so it would show up as reheat associated with the terminal unit in the proposed.  Please let us know if you disagree.

Reply:
@claperle so my thinking is as follows

1. First chunk of code (Lines 21-28) checks is a HVAC systems is present in P and U. If P has it and U doesnt, then it is a virtual system.
2. Seconf chunk of code (lines 29-36) checks if HVAC system which is present in both P and U has heating !=none in P but = None in U, in which case, it is a virtual system.
3. The 3rd chunk of code (lines 37-43) checks if terminal units are present in P which are not present in U or if terminal units with reheat in P are modeled without reheat in U. 

I believe the 3rd check is not required and confusing. If the intent is to check if virtual heating has been modeled then #1 and #2 are covering it. I don't quite understand why there would be a scenario where these is a HVAC system that has not been caught by #1 or #2. Essentially I don't understand what kind of scenario #3 would be applicable in. Essentially it will only apply in a scenario where both P and U have matching HVAC system, U and P have central heating but U doesn't have terminal H and P has terminal H. Is that a feasible scenario? And if yes, does the rule apply to such a scenario?

Reply:
@supriyagoel @mkarpman For me the question that determines whether #3 should be removed is whether System 5 and System 7 have a heating coil in the system itself or just reheat at the terminal. If they just have reheat at the terminal then I think there could be a scenario where both the U-RMR and P-RMR have no heating at the system level (checks #1 and #2) but the P-RMR has reheat at the terminal level (so need #3).